File: Python/Algorithms/1287.py
from math import ceil

class Solution:
    def findSpecialInteger(self, arr):
        length = len(arr)
        threshold, count = ceil(length * 0.25), 1
        for i in range(1, length):
            if arr[i-1] == arr[i]:
                count = count + 1
                if count > threshold:
                    return arr[i]
            else:
                count = 1
        return arr[0]

File: Rust/Algorithms/1287.rs
impl Solution {
    pub fn find_special_integer(arr: Vec<i32>) -> i32 {
        let threshold = ((arr.len() as f64) * 0.25) as i32;
        let mut count = 0;
        for i in 1..arr.len() {
            if arr[i - 1] == arr[i] {
                count += 1;
                if count > threshold {
                    return arr[i];
                }
            } else {
                count = 1;
            }
        }
        arr[0]
    }
}
